src/_guides/environment-declarations.md

```diff
@@ -0,0 +1,188 @@
+
+Your app can use the values of environment declarations
+to change its functionality or behavior.
+Dart compilers can take advantage of this to eliminate code
+no longer reachable due to the values specified, through "tree shaking".
+
+Some potential use cases for defining environment declarations are:
+
+* Adding functionality during debugging, such as enabling logging.
+* Creating separate flavors of your application.
+* Configuring application behavior, such as the port of an HTTP server.
+* Enabling an experimental mode of your application for testing.
+* Switching between testing and production backends.
+
+To specify an environment declaration
+to an application with the [`dart run`][] or [`dart compile`][] subcommands,
+use the `-D` or `--define` options and specify `<NAME>=<VALUE>`:
+
+```terminal
+$ dart run --define=DEBUG=true -DFLAVOR=free
+```
+
+To learn more about specifying environment declarations,
+including for other tools and compilers, see
+[Specifying environment declarations][].
+
+[`dart run`]: /tools/dart-run
+[`dart compile`]: /tools/dart-compile
+[Specifying environment declarations]: #specifying-environment-declarations
+
+## Accessing environment declarations
+
+To access specified environment declaration values,
+use one of the `fromEnvironment` constructors
+with `const` or within a constant context.
+Use [`bool.fromEnvironment`][bool-from] for `true` or `false` values,
+[`int.fromEnvironment`][int-from] for integer values,
+and [`String.fromEnvironment`][string-from] for anything else.
+
+{{site.alert.note}}
+  The environment declaration constructors are only guaranteed
+  to work when invoked as `const`.
+  Most compilers must be able to evaluate their value at compile time.
+{{site.alert.end}}
+
+Each of the `fromEnvironment` constructors require the
+name or key of the environment declaration key-value pair,
+as well as an optional `defaultValue` override
+that is used when a declaration wasn't defined
+or the specified value cannot be parsed as the expected type.
+
+For example, if you only want to log messages
+when the environment declaration `DEBUG` is set to `true`:
+
+<?code-excerpt "misc/lib/development/environment_declarations.dart (debug-log)"?>
+```dart
+void log(String message) {
+  // Log the debug message if the environment declaration 'DEBUG' is `true`.
+  // If there was no value specified, do not log.
+  if (const bool.fromEnvironment('DEBUG', defaultValue: false)) {
+    print('Debug: $message');
+  }
+}
+```
+
+In this snippet, if `DEBUG` is set to `false`
+during compilation, or not specified at all,
+production compilers can completely remove the condition and its body.
+
+The `fromEnvironment` constructors always
+have a default value for when the declaration isn't specified
+or the specified value cannot be parsed.
+Therefore, if you need to specifically check whether
+an environment declaration has been specified,
+use the [`bool.hasEnvironment`][bool-has] constructor:
```

Comment:
**issue:** Cut these sentences up. They're too long.

Reply:
I tried a few different ways of breaking them up, but most things I tried didn't read as well. I made a few other modifications to the text, but let me know if you have any ideas.